infrastructure/network/rpcclient/rpc_on_pruning_point_utxo_set_override.go

```diff
@@ -0,0 +1,57 @@
+package rpcclient
+
+import (
+	"github.com/kaspanet/kaspad/app/appmessage"
+	routerpkg "github.com/kaspanet/kaspad/infrastructure/network/netadapter/router"
+	"github.com/pkg/errors"
+)
+
+// RegisterPruningPointUTXOSetNotifications sends an RPC request respective to the function's name and returns the RPC server's response.
+// Additionally, it starts listening for the appropriate notification using the given handler function
+func (c *RPCClient) RegisterPruningPointUTXOSetNotifications(onPruningPointUTXOSetNotifications func()) error {
+
+	err := c.rpcRouter.outgoingRoute().Enqueue(appmessage.NewNotifyPruningPointUTXOSetOverrideRequestMessage())
+	if err != nil {
+		return err
+	}
+	response, err := c.route(appmessage.CmdNotifyPruningPointUTXOSetOverrideResponseMessage).DequeueWithTimeout(c.timeout)
+	if err != nil {
+		return err
+	}
+	notifyPruningPointUTXOSetOverrideResponse := response.(*appmessage.NotifyPruningPointUTXOSetOverrideResponseMessage)
+	if notifyPruningPointUTXOSetOverrideResponse.Error != nil {
+		return c.convertRPCError(notifyPruningPointUTXOSetOverrideResponse.Error)
+	}
+	spawn("RegisterPruningPointUTXOSetNotifications", func() {
+		for {
+			_, err := c.route(appmessage.CmdPruningPointUTXOSetOverrideNotificationMessage).Dequeue()
```

Comment:
Should we downcast the message to `PruningPointUTXOSetOverrideNotificationMessage` as a sanity check?